torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -2488,6 +2488,51 @@ Tensor linalg_qr_backward(const std::vector<torch::autograd::Variable> &grads, c
   }
 }
 
+// The backward for this function is just a specialized version of
+// lu.backward, which is implemented in /torch/_autograd_functions.py
+Tensor _det_lu_based_helper_backward(
+  const Tensor& det_grad,
+  const Tensor& self,
+  const Tensor& det,
+  const Tensor& p,
+  const Tensor& l,
+  const Tensor& u
+) {
+  if (!det_grad.defined()) {
+    return at::zeros_like(self, at::MemoryFormat::Contiguous);
+  }
+  auto u_h = u.transpose(-2, -1).conj();
+  auto u_h_diag = u_h.diagonal(0, -2, -1);
+  auto u_h_conditioned = at::where(
```

Comment:
When I was working on `det_backward`, I had an issue with `at::where` not supporting complex numbers: https://github.com/pytorch/pytorch/pull/54787#discussion_r602532789

Has support been added since then?

Reply:
Ah it has been added, nevermind.